apps/explorer/lib/explorer/chain.ex

```diff
@@ -4633,7 +4638,7 @@ defmodule Explorer.Chain do
         query
         |> limit(^(page_size + 1))
 
-      page_in_bounds?(page_number, page_size) ->
+      page_in_bounds?(page_number, page_size) || no_page_limit ->
```

Comment:
Looking at this again, it will probably make more sense to add the logic in `page_in_bounds?`.
I will approve for now